agent/ecscni/plugin.go

```diff
@@ -0,0 +1,186 @@
+	if err != nil {
+		return err
+	}
+
+	seelog.Debugf("Set up container namespace done: %v", result)
+	return nil
+}
+
+// CleanupNS will clean up the container namespace, including remove the veth
+// pair, release the ip address in ipam
+func (client *cniClient) CleanupNS(cfg *Config) error {
+	cns := &libcni.RuntimeConf{
+		ContainerID: cfg.ContainerID,
+		NetNS:       fmt.Sprintf(netnsFormat, cfg.ContainerPID),
+		IfName:      defaultEthName,
+	}
+
+	netConfigList, err := client.constructNetworkConfig(cfg)
+	if err != nil {
+		return errors.Wrap(err, "cni invocation: Failed to construct network configuration to clean up namespace")
+	}
+
+	seelog.Debugf("Starting clean up the container namespace: %s", cfg.ContainerID)
+	return client.libcni.DelNetworkList(netConfigList, cns)
+}
+
+// constructNetworkConfig creates configuration for eni, ipam and bridge plugin
+func (client *cniClient) constructNetworkConfig(cfg *Config) (*libcni.NetworkConfigList, error) {
+	_, dst, err := net.ParseCIDR(TaskIAMRoleEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	ipamConf := IPAMConfig{
+		Type:        "ipam",
+		CNIVersion:  client.cniVersion,
+		IPV4Subnet:  client.subnet,
+		IPV4Address: cfg.IPAMV4Address,
+		ID:          cfg.ID,
+		IPV4Routes: []*types.Route{
+			{
+				Dst: *dst,
+			},
+		},
+	}
+
+	bridgeName := defaultBridgeName
+	if cfg.BridgeName != "" {
+		bridgeName = cfg.BridgeName
+	}
+	bridgeConf := BridgeConfig{
+		Type:       "bridge",
+		CNIVersion: client.cniVersion,
+		BridgeName: bridgeName,
+		IsGW:       true,
+		IPAM:       ipamConf,
+	}
+
+	eniConf := ENIConfig{
+		Type:        "eni",
+		CNIVersion:  client.cniVersion,
+		ENIID:       cfg.ENIID,
+		IPV4Address: cfg.ENIIPV4Address,
+		MACAddress:  cfg.ENIMACAddress,
+		IPV6Address: cfg.ENIIPV6Address,
+	}
+
+	bridgeConfBytes, err := json.Marshal(bridgeConf)
+	if err != nil {
+		seelog.Errorf("Marshal bridge configuration failed, error: %v", err)
+		return nil, err
+	}
+	plugins := []*libcni.NetworkConfig{
+		&libcni.NetworkConfig{
+			Network: &types.NetConf{
+				Type: "bridge",
+			},
+			Bytes: bridgeConfBytes,
+		},
+	}
+
+	eniConfBytes, err := json.Marshal(eniConf)
+	if err != nil {
+		seelog.Errorf("Marshal eni configuration error: %v", err)
+		return nil, err
+	}
+	plugins = append(plugins, &libcni.NetworkConfig{
+		Network: &types.NetConf{
+			Type: "eni",
+		},
+		Bytes: eniConfBytes,
+	})
+
+	netconf := &libcni.NetworkConfigList{
+		Name:       NetworkName,
+		CNIVersion: client.cniVersion,
+		Plugins:    plugins,
+	}
+	return netconf, nil
+}
+
+// Version returns the version of the plugin
+func (client *cniClient) Version(name string) (string, error) {
+	file := filepath.Join(client.pluginsPath, name)
+
+	// Check if the plugin execute file exists
+	_, err := os.Stat(file)
+	if err != nil {
+		return "", err
+	}
+
+	cmd := exec.Command(file, VersionCommand)
+	versionInfo, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+
+	version := &struct {
```

Comment:
Can you break this out (lines 177-185) into a method so that we can write unit tests?